keps/sig-node/3573-device-plugin/README.md

```diff
@@ -493,12 +614,199 @@ Follow protobuf guidelines on versioning:
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+* A rollback to Kubernetes version <1.10 can impact running workloads as they require
+  API version to exactly match for Kubelet and Device Plugin.
+
+* A rollout can fail in case a bug is introduced either in device manager or device
+  plugin preventing already running pods from restarting or new pods to start.
+
+###### What specific metrics should inform a rollback?
+
+Not Applicable, metrics specific to rollback are not available.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Yes
+The device plugin API is stable since its graduation to Beta and device plugins using
+`v1beta1` API are compatible with any Kuberntes version >= 1.10. Therefore, ugrades or
+downgrades don't have an impact as long as the upgraded/downgraded version is > Kubernetes
+v1.10.
 
+When upgrading device plugins, it is recommended that the cluster admin drain workloads
+requesting devices and conform to the device plugin API.
+
+This has been validated by several device plugins. Here is an example: https://github.com/NVIDIA/k8s-device-plugin#upgrading-kubernetes-with-the-device-plugin. 
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+1. Kubelet PodResource API endpoint can be used to obtain information on devices
+   allocated to running pods. Refer to [keps/sig-node/606-compute-device-assignment]([keps/sig-node/606-compute-device-assignment])
+2. Status of the nodes  specifically device capacity and allocatable can be inspected
+   to see if device plugins were successflly deployed and are avaialable to be allocated
+   on the node.
+   In addition to that, status of the running pods can be inspected to determine if the
+   pod requesting device(s) was running successfully and allocated device(s) as per its
+   request.
+3. Typically, device allocation to a workload is accompanied by creation of environment
+   variable capturing device specification identification information e.g. PCI address
+   information which can be inspected, creation of device files and device specific
+   volume mount points.This varies depending on the device plugin implementation. 
+
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] API .status
+  - Field: Node.status
+    - Property: [capacity,allocatable]
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Not Applicable, refer to Kubelet SLOs.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name: `pod_resources_endpoint_requests_total`
```

Comment:
Unfortunately I'm not sure this metric is helping here because it tracks the total time the podresources endpoints were queried. IOW, the metric will tell how many time `List` + `GetAllocatable` have been called

Reply:
I believe you are right, I will remove it. Thanks